numpy/testing/nose_tools/utils.py

```diff
@@ -576,8 +576,19 @@ def _build_err_msg():
             return
     except (NotImplementedError, TypeError):
         pass
-    if abs(desired - actual) >= 1.5 * 10.0**(-decimal):
-        raise AssertionError(_build_err_msg())
+
+    try:
+        if abs(desired - actual) >= 1.5 * 10.0**(-decimal):
+            raise AssertionError(_build_err_msg())
+    except TypeError:
+        # Certain types can not be subtracted (numpy.bool_ for example)
```

Comment:
Sure it's not better to just fall back on `assert_equal` here?

Reply:
I think not since a tolerance is either provided by the user or at least implied by the "almost." Indeed, `True` is almost equal `False` when `decimal = 0`. Also, this should work for any type that can not be exactly compared but can be converted to a float-like type (result of `result_type(value, 1.)`)

Reply:
I'd argue that "almost" has no meaning for booleans, and that there is no level of "almost" for which `True ~= False`

Reply:
I agree that using this method with booleans is probably not useful. However this patch allows for *any type* that doesn't implement subtract but can be converted to `float` - bools included.

Additionally, it brings `assert_almost_equal` in-line with the behavior of `assert_array_almost_equal` which converts the array dtype using `result_type(value, 1.)`.

Reply:
> However this patch allows for any type that doesn't implement 

I'd go as far as saying that "almost" is only meaningful for types that define subtraction. Consistency with `assert_array_almost_equal` is obviously good, but perhaps we should change `assert_almost_equal ` instead

Reply:
We'll need a third opinion. I could go either way, but the counter argument goes like this: "almost" is meaningful for any object/type that is or can be converted to a numeric type.

Reply:
By that argument, is `almost_equal("1.0001", "1")`?  (and does this patch change that?)

But sure, a third opinion wouldn't do any harm